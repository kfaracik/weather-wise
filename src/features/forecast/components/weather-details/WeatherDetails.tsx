import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Text} from 'react-native-paper';
import {BaseWeatherData} from '../../api/types';

type WeatherDetailsProps = {
  data: BaseWeatherData;
};

export const WeatherDetails = ({data}: WeatherDetailsProps) => (
  <View style={styles.container}>
    <Text style={styles.title}>Weather Details</Text>
    <Text style={styles.detail}>Humidity: {data.main.humidity}%</Text>
    <Text style={styles.detail}>Pressure: {data.main.pressure} hPa</Text>
    <Text style={styles.detail}>Wind Speed: {data.wind.speed} m/s</Text>
    <Text style={styles.detail}>Cloudiness: {data.clouds.all}%</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    marginTop: 16,
    marginBottom: 32,
    padding: 16,
    borderRadius: 8,
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 8,
  },
  detail: {
    fontSize: 16,
    marginBottom: 4,
  },
});
