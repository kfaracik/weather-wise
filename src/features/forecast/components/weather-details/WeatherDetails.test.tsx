import React from 'react';
import {render} from '@testing-library/react-native';
import {WeatherDetails} from './WeatherDetails';

describe('WeatherDetails', () => {
  it('should generate snapshot', () => {
    const {toJSON} = render(
      <WeatherDetails
        data={{
          dt: 0,
          main: {
            temp: 40,
            feels_like: undefined,
            temp_min: 40,
            temp_max: 30,
            pressure: 20,
            humidity: 10,
            sea_level: undefined,
            grnd_level: undefined,
            temp_kf: undefined,
          },
          weather: [],
          clouds: {
            all: 60,
          },
          wind: {
            speed: 20,
            deg: undefined,
            gust: undefined,
          },
          visibility: 30,
          pop: undefined,
          rain: undefined,
          snow: undefined,
          sys: {
            country: 'PL',
            sunrise: undefined,
            sunset: undefined,
            pod: undefined,
          },
          dt_txt: undefined,
        }}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
