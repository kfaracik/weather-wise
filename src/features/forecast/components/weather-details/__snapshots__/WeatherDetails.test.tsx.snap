// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetails should generate snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255, 255, 255, 0.5)",
      "borderRadius": 8,
      "marginBottom": 32,
      "marginTop": 16,
      "padding": 16,
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 8,
        },
      ]
    }
  >
    Weather Details
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 16,
          "marginBottom": 4,
        },
      ]
    }
  >
    Humidity: 
    10
    %
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 16,
          "marginBottom": 4,
        },
      ]
    }
  >
    Pressure: 
    20
     hPa
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 16,
          "marginBottom": 4,
        },
      ]
    }
  >
    Wind Speed: 
    20
     m/s
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 16,
          "marginBottom": 4,
        },
      ]
    }
  >
    Cloudiness: 
    60
    %
  </Text>
</View>
`;
